packages/mui-base/src/OptionUnstyled/optionUnstyledClasses.tsx

```diff
@@ -2,9 +2,13 @@ import generateUtilityClass from '../generateUtilityClass';
 import generateUtilityClasses from '../generateUtilityClasses';
 
 export interface OptionUnstyledClasses {
+  /** Class name applied to the root element. */
   root: string;
+  /** State class applied to the root `li` element if `disabled={true}`. */
   disabled: string;
+  /** State class applied to the root `li` element if `selected={true}`. */
   selected: string;
+  /** State class applied to the root `li` element if `highlighted={true}`. */
```

Comment:
Should we indicate somehow in the description that some of these state classes are global. For e.g. disabled would be `Mui-disabled`, while highlighted would be `MuiOption-highlighted`. We don't do this distinction in Material UI, so it's not pressing, but probably is worth at least to consider this. cc @samuelsycamore

Reply:
Will this be clear enough?

e.g.,
```
/** State class applied to the root `li` element if `highlighted={true}`. */
highlighted?: string;
```

```
/** Global state class applied to the root element if `disabled={true}`. */
disabled: string;
```

Reply:
If we worry that developers won't know the distinction, in the Classes tables (https://deploy-preview-36589--material-ui.netlify.app/base/api/slider-unstyled/#classes, https://deploy-preview-36589--material-ui.netlify.app/material-ui/api/slider/#css), we can make some distinction between the two, e.g., highlighting the global state classes.

Reply:
Ah nice idea, I like this. @samuelsycamore what do you think?

Reply:
We can continue this in a separate PR after Sam replies. I will merge this PR.

Reply:
Made a PR for this: https://github.com/mui/material-ui/pull/36633

Reply:
Looks good to me! I think the distinction between "global" and "component" state classes is helpful and clear.